File: Python3/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
from collections import Counter

def solution(array):
    counter = Counter(array)
    
    max_count = max(counter.values())
    
    max_count_occurences = list(counter.values()).count(max_count)
    
    if max_count_occurences > 1:
        return -1
    
    for num, count in counter.items():
        if count == max_count:
            return num
        
    return -1

File: Swift/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.swift
import Foundation

func solution(_ array:[Int]) -> Int {
    var countDict: [Int: Int] = [:]
    
    for num in array {
        countDict[num, default: 0] += 1
    }
    
    let maxCount = countDict.values.max() ?? -1

    let maxCountOccurrences = countDict.values.filter { $0 == maxCount }.count

    if maxCountOccurrences > 1 {
        return -1
    }
    
    for (key, value) in countDict {
        if value == maxCount {
            return key
        }
    }
    
    return -1
}